src/compiler.jl

```diff
@@ -304,15 +304,18 @@ function runtime_generic_rev(fn::Any, ret_ptr::Ptr{Any}, arg_ptr::Ptr{Any}, shad
 
     # TODO handle active args
     tt′   = Tuple{map(Core.Typeof, args)...}
-    ptr   = Compiler.deferred_codegen(Val(fn), Val(tt′))
     tt    = Tuple{map(T->eltype(Core.Typeof(T)), args)...}
     rt    = Core.Compiler.return_type(fn, tt)
 
     if rt <: AbstractFloat || rt <: Complex{<:AbstractFloat}
         push!(args, tape)
+        rt = Active{rt}
+    else
```

Comment:
This is too aggressive and should probably be extracted into its own function. Integers, ghost types, etc can be const by default, boxed values/arrays are duplicated, etc.

Reply:
We don't support Duplicated as a return activity yet, right?